guidelines/mwm-auth.md

```diff
@@ -0,0 +1,86 @@
+<!--
+  ~ Copyright 2022 MONAI Consortium
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+-->
+
+# Authentication notes
+
+In the developement of the authentication we tested with Keycloak to note you need to flatten the roles for the authentication to be able to pick up the roles this can be done by going to 
+
+Keycloak -> Admin dashboard -> your specific realm -> Clients -> Mappers -> Create
```

Comment:
Sorry for the naive question, are we assuming that organizations will be creating a realm specifically for MONAI to have this setting, so it doesn't conflict with other applications?

Reply:
No the JWToken just needs to match rules in config this is just how it was setup in development.
We will accept any JWToken provider so if Okta providers the token for example okta needs to provide it in a flat format and you'd put field your expecting token in and its expected value.

![image](https://user-images.githubusercontent.com/61380713/185577553-bbdc512e-31b3-4454-afc5-0cb7e7760fc4.png)

for example in this picture the blue arrow is the way keycloak providers roles is a object inside object like...
resource_access.monai-app-test.roles 

Which asp.net doesn't handle very nicely you can see resource_access but then you'd have to deserialise the rest of it.

but if you create this mapper specified above it will be a flat entry just "user_roles".contains("monai-role-admin")